service/common/src/main/java/org/apache/polaris/service/catalog/BasePolarisCatalog.java

```diff
@@ -1254,6 +1267,7 @@ public void doRefresh() {
       if (latestLocation == null) {
         disableRefresh();
       } else {
+        polarisEventListener.onBeforeRefreshTable(new BeforeTableRefreshEvent(tableIdentifier));
```

Comment:
This approach of manually triggering events is fine as a starting point, but I think a more elegant approach would be to create a decorator catalog that would decorate calls to each method with before/after events.

Reply:
Yea that would be interesting. It might not be ideal to couple events to methods though, as that could force you to unnecessarily extract things into a separate method.